src/Core.UnitTests/Notifications/DisabledNotificationsStorageTests.cs

```diff
@@ -0,0 +1,136 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Core.Notifications;
+using SonarLint.VisualStudio.Core.VsVersion;
+
+namespace SonarLint.VisualStudio.Core.UnitTests.Notifications
+{
+    [TestClass]
+    public class DisabledNotificationsStorageTests
+    {
+        [DataRow("2", true)]
+        [DataRow("4", false)]
+        [TestMethod]
+        public void IsNotificationDisabled_ReturnsCorrect(string notificationId, bool expectedResult)
+        {
+            var file = CreateFileMock("1", "2", "3");
+            var testSubject = CreateTestSubject(file.Object);
+                        
+            var result = testSubject.IsNotificationDisabled(notificationId);
+
+            result.Should().Be(expectedResult);
+            file.Verify(f => f.ReadAllLines(GetFilePath()), Times.Once);
+        }
+
+        [TestMethod]
+        public void DisableNotification_IsNotDisabled_Disables()
+        {
+            var file = CreateFileMock("1", "2", "3");
+            var testSubject = CreateTestSubject(file.Object);
+
+            var result = testSubject.IsNotificationDisabled("4"); 
+            result.Should().BeFalse();
+
+            testSubject.DisableNotification("4");
```

Comment:
None of the tests check what happens if you call Disable without calling IsDisabled first

Reply:
What I meant was that IsDisabled should be tested in isolation.